node/src/run_modes.rs

```diff
@@ -115,185 +197,153 @@ impl RunModes {
+    daemon_initializer_factory: Box<dyn DaemonInitializerFactory>,
+}
 
 impl Runner for RunnerReal {
-    fn run_service(
-        &self,
-        args: &[String],
-        streams: &mut StdStreams<'_>,
-    ) -> Result<i32, ConfiguratorError> {
+    fn run_node(&self, args: &[String], streams: &mut StdStreams<'_>) -> Result<(), RunnerError> {
         let system = System::new("main");
-
-        let mut server_initializer = ServerInitializer::new();
-        server_initializer.go(streams, args);
+        let mut server_initializer = self.server_initializer_factory.make();
+        server_initializer.go(streams, args)?;
 
         actix::spawn(server_initializer.map_err(|_| {
             System::current().stop_with_code(1);
         }));
-
-        Ok(system.run())
+        match system.run() {
+            0 => Ok(()),
+            num_e => Err(RunnerError::Numeric(num_e)),
+        }
     }
 
     fn dump_config(
         &self,
         args: &[String],
         streams: &mut StdStreams<'_>,
-    ) -> Result<i32, ConfiguratorError> {
-        config_dumper::dump_config(args, streams)
+    ) -> Result<(), RunnerError> {
+        self.dump_config_runner_factory
+            .make()
+            .go(streams, args)
+            .map_err(RunnerError::Configurator)
     }
 
-    fn initialization(
-        &self,
-        args: &[String],
-        streams: &mut StdStreams<'_>,
-    ) -> Result<i32, ConfiguratorError> {
-        let configurator = NodeConfiguratorInitialization {};
-        let config = configurator.configure(args, streams)?;
-        let mut initializer = DaemonInitializer::new(
-            &RealDirsWrapper {},
-            Box::new(LoggerInitializerWrapperReal {}),
-            config,
-            Box::new(ChannelFactoryReal::new()),
-            Box::new(RecipientsFactoryReal::new()),
-            Box::new(RerunnerReal::new()),
-        );
-        initializer.go(streams, args);
-        Ok(1)
+    fn run_daemon(&self, args: &[String], streams: &mut StdStreams<'_>) -> Result<(), RunnerError> {
+        let mut initializer = self.daemon_initializer_factory.make(args, streams)?;
+        initializer.go(streams, args)?;
+        Ok(()) //there might presently be no way to make this fn terminate politely
     }
+}
 
-    fn configuration_run(
-        &self,
-        args: &[String],
-        streams: &mut StdStreams<'_>,
-        configurator: &dyn NodeConfigurator<WalletCreationConfig>,
-        _privilege_dropper: &dyn PrivilegeDropper,
-    ) -> Result<i32, ConfiguratorError> {
-        match configurator.configure(args, streams) {
-            Ok(_) => Ok(0),
-            Err(e) => Err(e),
+impl RunnerReal {
+    pub fn new() -> Self {
+        Self {
+            dump_config_runner_factory: Box::new(DumpConfigRunnerFactoryReal),
+            server_initializer_factory: Box::new(ServerInitializerFactoryReal),
+            daemon_initializer_factory: Box::new(DaemonInitializerFactoryReal::build()),
```

Comment:
I was expecting to see either ::default() or ::new() here...

Reply:
I incline to keeping a nonstandard name because default() evokes that it's something static, but I create have to build the factory first and since it's more complex than usually I will stick with a compromise: "build_factory()"